### PR TITLE
[docs-only] Add tus upload documentation

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR add document related to api requestion to make tus upload

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [x] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by mmattel** [comment]
@EParzefall mind to have a look on it for a language review?

File: docs/apis/http/tus_upload.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+title: "TUS upload"
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:2]:
```suggestion
title: "tus Upload"
```

Comment (EParzefall) [reply docs/apis/http/tus_upload.md:2]: This is how tus is spelled on https://tus.io/ so I guess, we should use that spelling, even at the beginning of sentences.

File: docs/apis/http/tus_upload.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+date: 2023-10-10T00:00:00+00:00
+weight: 21
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/apis/http
+geekdocFilePath: tus_upload.md
+geekdocCollapseSection: true
+---
+
+Infinite Scale supports TUS upload protocols.
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:11]:
```suggestion
Infinite Scale supports tus upload protocols.
```

File: docs/apis/http/tus_upload.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+Tus is a robust, modular, and open protocol designed to resume large file uploads reliably over HTTP.
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:12]:
```suggestion
tus is a robust, modular, and open protocol designed to resume large file uploads reliably over HTTP.
```

File: docs/apis/http/tus_upload.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+In situations where file uploads might be interrupted due to network issues, browser crashes, or other unforeseen interruptions,
+Tus ensures that uploads can be resumed from the point of failure without losing data. Read details about [TUS protocol](https://tus.io/protocols/resumable-upload)
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:14]:
```suggestion
tus ensures that uploads can be resumed from the point of failure without losing data. Read details about [TUS protocol](https://tus.io/protocols/resumable-upload)
```

File: docs/apis/http/tus_upload.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+## Create an upload URL
+The Client must send a POST request against a known upload creation URL to request a new upload resource.
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:17]:
```suggestion
The client must send a POST request against a known upload creation URL to request a new upload resource.
```

File: docs/apis/http/tus_upload.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+You need to provide filename in base64 encoded format
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:18]:
```suggestion
You need to provide a file name in base64-encoded format
```

File: docs/apis/http/tus_upload.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+{{< /tab >}}
+{{< /tabs >}}
+
+The server will return a temporary upload URL in the location header of the response:
+```bash
+< Location: https://localhost:9200/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNjk3NTMyNTc5LCJpYXQiOjE2OTc0NDYxNzksInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy8zYTU3ZWZlMS04MzE0LTQ4MGEtOWY5Ny04N2Q1YzBjYTJhMTgifQ.FbrlY7mdOfsbFgMrP8OtcHlCEq72a2ZVnPD2iBo9MfM
+```
+
+## Upload a file
+Once a temporary upload URL has been created, a client can send a PATCH request to upload a file. The file content should be send in the body of the request:
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:65]:
```suggestion
Once a temporary upload URL has been created, a client can send a PATCH request to upload a file. The file content should be sent in the body of the request:
```

File: docs/apis/http/tus_upload.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+The server will return a temporary upload URL in the location header of the response:
+```bash
+< Location: https://localhost:9200/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNjk3NTMyNTc5LCJpYXQiOjE2OTc0NDYxNzksInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy8zYTU3ZWZlMS04MzE0LTQ4MGEtOWY5Ny04N2Q1YzBjYTJhMTgifQ.FbrlY7mdOfsbFgMrP8OtcHlCEq72a2ZVnPD2iBo9MfM
+```
+
+## Upload a file
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:64]:
```suggestion
## Upload a File
```

File: docs/apis/http/tus_upload.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+geekdocFilePath: tus_upload.md
+geekdocCollapseSection: true
+---
+
+Infinite Scale supports TUS upload protocols.
+Tus is a robust, modular, and open protocol designed to resume large file uploads reliably over HTTP.
+In situations where file uploads might be interrupted due to network issues, browser crashes, or other unforeseen interruptions,
+Tus ensures that uploads can be resumed from the point of failure without losing data. Read details about [TUS protocol](https://tus.io/protocols/resumable-upload)
+
+## Create an upload URL
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:16]:
```suggestion
## Create an Upload URL
```

File: docs/apis/http/tus_upload.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+{{< /tabs >}}
+
+The server will return a temporary upload URL in the location header of the response:
+```bash
+< Location: https://ocis.test/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNjk3NTMyNTc5LCJpYXQiOjE2OTc0NDYxNzksInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy8zYTU3ZWZlMS04MzE0LTQ4MGEtOWY5Ny04N2Q1YzBjYTJhMTgifQ.FbrlY7mdOfsbFgMrP8OtcHlCEq72a2ZVnPD2iBo9MfM
+```
+
+## Upload the first chunk
+Once a temporary upload URL has been created, a client can send a PATCH request to upload a file. The file content should be sent in the body of the request:
+{{< tabs "create-upload" >}}
```
Comment (KarunAtreya) [inline docs/apis/http/tus_upload.md:68]:
```suggestion
{{< tabs "upload-the-first-chunk" >}}
```

File: docs/apis/http/tus_upload.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+< X-Request-Id: amrita-OptiPlex-3090/LOcqGy6p2r-000381
+<
+* Connection #0 to host localhost left intact
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Upload the second chunk
+After first chunk uploaded, second chunk can be uploaded with pointing `Upload-Offset` to exact location
+{{< tabs "create-upload" >}}
```
Comment (KarunAtreya) [inline docs/apis/http/tus_upload.md:96]:
```suggestion
{{< tabs "upload-the-second-chunk" >}}
```

File: docs/apis/http/tus_upload.md [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+```
+{{< /tab >}}
+{{< /tabs >}}
+{{< hint type=warning title="Important Warning" >}}
+`Upload-Offset` header indicates the byte position in the target file where the server should start writing the upload content.
+It ensures data integrity and order during the upload process.
+{{< /hint >}}
+
+# Creation with Upload
+{{< tabs "create-upload" >}}
```
Comment (KarunAtreya) [inline docs/apis/http/tus_upload.md:127]:
```suggestion
{{< tabs "creation-with-upload" >}}
```

File: docs/apis/http/tus_upload.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+< X-Content-Type-Options: nosniff
+< X-Request-Id: amrita-OptiPlex-3090/LOcqGy6p2r-000381
+<
+* Connection #0 to host localhost left intact
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Upload the second chunk
+After first chunk uploaded, second chunk can be uploaded with pointing `Upload-Offset` to exact location
```
Comment (KarunAtreya) [inline docs/apis/http/tus_upload.md:95]:
```suggestion
After the first chunk is uploaded, second chunk can be uploaded by pointing `Upload-Offset` to exact position that was returned in the first response
```

File: docs/apis/http/tus_upload.md [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+geekdocFilePath: tus_upload.md
+geekdocCollapseSection: true
+---
+
+Infinite Scale supports [tus resumable-upload protocol](https://tus.io/protocols/resumable-upload).
+tus is a robust, modular, and open protocol designed to resume large file uploads reliably over HTTP.
+In situations where file uploads might be interrupted due to network issues, browser crashes, or other unforeseen interruptions,
+tus ensures that uploads can be resumed from the point of failure without losing data.
+
+# Upload in Chunks
```
Comment (micbar) [inline docs/apis/http/tus_upload.md:16]: please start with a level 2 heading, because the document title is already level 1.

**Comment by micbar** [comment]
Good writeup in general.

File: docs/apis/http/tus_upload.md [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+The server will return a temporary upload URL in the location header of the response:
+```bash
+< Location: https://ocis.test/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNjk3NTMyNTc5LCJpYXQiOjE2OTc0NDYxNzksInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy8zYTU3ZWZlMS04MzE0LTQ4MGEtOWY5Ny04N2Q1YzBjYTJhMTgifQ.FbrlY7mdOfsbFgMrP8OtcHlCEq72a2ZVnPD2iBo9MfM
+```
+
+## Upload the first chunk
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:66]:
```suggestion
## Upload the First Chunk
```

File: docs/helpers/extended_vars.yaml [hunk 16]
```diff
@@ -39,8 +39,8 @@ variables:
   type: string
   default_value: ""
   description: 'Go micro registry type to use. Supported types are: ''nats'', ''kubernetes'',
-    ''etcd'', ''consul'', ''memory'' and ''mdns''. Will be selected automatically. Only change
-    on supervision of ownCloud Support.'
+    ''etcd'', ''consul'', ''memory'' and ''mdns''. Will be selected automatically.
+    Only change on supervision of ownCloud Support.'
```
Comment (EParzefall) [inline docs/helpers/extended_vars.yaml:43]:
```suggestion
    Only change under supervision of ownCloud Support.'
```

File: docs/apis/http/tus_upload.md [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+< X-Content-Type-Options: nosniff
+< X-Request-Id: amrita-OptiPlex-3090/LOcqGy6p2r-000381
+<
+* Connection #0 to host localhost left intact
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Upload the second chunk
+After the first chunk is uploaded, second chunk can be uploaded by pointing `Upload-Offset` to exact position that was returned in the first response
```
Comment (EParzefall) [inline docs/apis/http/tus_upload.md:95]:
```suggestion
After the first chunk is uploaded, the second chunk can be uploaded by pointing `Upload-Offset` to exact position that was returned in the first response
```

**Review by EParzefall — COMMENTED** [review]
LGTM

File: docs/apis/http/tus_upload.md [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+title: "tus upload"
+date: 2023-10-10T00:00:00+00:00
+weight: 21
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/apis/http
+geekdocFilePath: tus_upload.md
+geekdocCollapseSection: true
+---
+
+Infinite Scale supports [tus resumable-upload protocol](https://tus.io/protocols/resumable-upload).
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:11]:
```suggestion
ownCloud Infinite Scale supports [tus resumable-upload protocol](https://tus.io/protocols/resumable-upload).
```

File: docs/apis/http/tus_upload.md [hunk 19]
```diff
@@ -0,0 +1,178 @@
...
+tus is a robust, modular, and open protocol designed to resume large file uploads reliably over HTTP.
+In situations where file uploads might be interrupted due to network issues, browser crashes, or other unforeseen interruptions,
+tus ensures that uploads can be resumed from the point of failure without losing data.
+
+## Upload in Chunks
+### Create an Upload URL
+The client must send a POST request against a known upload creation URL to request a new upload resource.
+You need to provide filename in base64-encoded format.
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:19]:
```suggestion
The client must send a POST request against a known upload creation URL to request a new upload resource.
The filename has to be provided in base64-encoded format.
```

File: docs/apis/http/tus_upload.md [hunk 20]
```diff
@@ -0,0 +1,178 @@
...
+
+Example:
+```bash
+'tustest.txt' = 'dHVzdGVzdC50eHQ=' //encoded in base64 format
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:23]:
```suggestion
# base64 encoded filename 'tustest.txt' is 'dHVzdGVzdC50eHQ='
echo -n 'tustest.txt' | base64
```

File: docs/apis/http/tus_upload.md [hunk 21]
```diff
@@ -0,0 +1,178 @@
...
+```
+{{< tabs "create-upload-url" >}}
+{{< tab "Request" >}}
+```bash
+curl -ks -u admin:admin -XPOST https://ocis.test/remote.php/dav/spaces/8d72036d-14a5-490f-889e-414064156402$196ac304-7b88-44ce-a4db-c4becef0d2e0 \
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:28]: let's not use basic auth, because the is disabled by default

File: docs/apis/http/tus_upload.md [hunk 22]
```diff
@@ -0,0 +1,178 @@
...
+-H "Tus-Resumable: 1.0.0" \
+-H "Upload-Length: 10" \
+-H "Upload-Metadata: filename dHVzdGVzdC50eHQ="
+```
+{{< /tab >}}
+{{< tab "Response - 201 Created" >}}
+```xmllint --format
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:35]: why is this line here?

File: docs/apis/http/tus_upload.md [hunk 23]
```diff
@@ -0,0 +1,178 @@
...
+< X-Xss-Protection: 1; mode=block
+<
+* Connection #0 to host localhost left intact
+
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+The server will return a temporary upload URL in the location header of the response:
+```bash
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:62]:
```suggestion
```
```

File: docs/apis/http/tus_upload.md [hunk 24]
```diff
@@ -0,0 +1,178 @@
...
+< Vary: Origin
+< X-Content-Type-Options: nosniff
+< X-Request-Id: xxxxxxxxxxxxxxxxxxxxxx
+<
+* Connection #0 to host localhost left intact
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+### Upload the Second Chunk
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:94]:
```suggestion
### Upload Further Chunks
```

File: docs/apis/http/tus_upload.md [hunk 25]
```diff
@@ -0,0 +1,178 @@
...
+<
+* Connection #0 to host localhost left intact
+```
+{{< /tab >}}
+{{< /tabs >}}
+{{< hint type=warning title="Important Warning" >}}
+`Upload-Offset` header indicates the byte position in the target file where the server should start writing the upload content.
+It ensures data integrity and order during the upload process.
+{{< /hint >}}
+
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:125]: maybe mention also that the upload will be finished when all bytes that were mentioned while creating the temporary URL are uploaded ( Header: `Upload-Length: 10"`)

File: docs/apis/http/tus_upload.md [hunk 26]
```diff
@@ -0,0 +1,178 @@
...
+< X-Frame-Options: SAMEORIGIN
+* TLSv1.2 (IN), TLS header, Supplemental data (23):
+{ [5 bytes data]
+< X-Permitted-Cross-Domain-Policies: none
+< X-Request-Id: xxxxxxxxxxxxxxxxxxxxxx
+< X-Robots-Tag: none
+< X-Xss-Protection: 1; mode=block
+<
+* Connection #0 to host localhost left intact
+-:1: parser error : Document is empty
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:171]: is this some left over?

File: docs/apis/http/tus_upload.md [hunk 27]
```diff
@@ -0,0 +1,178 @@
...
+geekdocEditPath: edit/master/docs/apis/http
+geekdocFilePath: tus_upload.md
+geekdocCollapseSection: true
+---
+
+Infinite Scale supports [tus resumable-upload protocol](https://tus.io/protocols/resumable-upload).
+tus is a robust, modular, and open protocol designed to resume large file uploads reliably over HTTP.
+In situations where file uploads might be interrupted due to network issues, browser crashes, or other unforeseen interruptions,
+tus ensures that uploads can be resumed from the point of failure without losing data.
+
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:15]: please mention that this documentation shows some basic examples and that more details about the tus protocol can be found in their own docs

**Review by individual-it — CHANGES_REQUESTED** [review]

File: docs/apis/http/tus_upload.md (hunk 18 above)
Comment (EParzefall) [reply docs/apis/http/tus_upload.md:11]: ownCloud should not be part of the product name, simply use Infinite Scale.

File: docs/apis/http/tus_upload.md [hunk 28]
```diff
@@ -0,0 +1,189 @@
...
+-H "Upload-Length: 14" \
+-H "Content-Type: application/offset+octet-stream" \
+-H "Upload-Metadata: filename dGVzdC50eHQ=" \
+-H "Tus-Extension: creation-with-upload" \
+-d "upload content"
+```
+{{< /tab >}}
+
+{{< tab "Response - 201 Created" >}}
+```xmllint --format
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:152]: what is the `xmllint --format` for? Is that a valid format?

File: docs/apis/http/tus_upload.md [hunk 29]
```diff
@@ -0,0 +1,189 @@
...
+{{< hint type=warning title="Important Warning" >}}
+`Upload-Offset` header indicates the byte position in the target file where the server should start writing the upload content.
+It ensures data integrity and order during the upload process.
+{{< /hint >}}
+
+## Creation with Upload
+{{< tabs "creation-with-upload" >}}
+{{< tab "Request" >}}
+```shell
+curl -ks -XPOST https://ocis.test/webdav/ \
```
Comment (individual-it) [inline docs/apis/http/tus_upload.md:140]: let's use a spaces path here too

**Review by individual-it — CHANGES_REQUESTED** [review]

File: docs/apis/http/tus_upload.md [hunk 30]
```diff
@@ -0,0 +1,189 @@
+---
+title: "tus upload"
```
Comment (mmattel) [inline docs/apis/http/tus_upload.md:2]:
```suggestion
title: "tus Upload"
```

File: docs/apis/http/tus_upload.md [hunk 31]
```diff
@@ -0,0 +1,189 @@
...
+date: 2023-10-10T00:00:00+00:00
+weight: 21
+geekdocRepo: https://github.com/owncloud/ocis
+geekdocEditPath: edit/master/docs/apis/http
+geekdocFilePath: tus_upload.md
+geekdocCollapseSection: true
+---
+
+Infinite Scale supports tus resumable-upload protocol.
```
Comment (mmattel) [inline docs/apis/http/tus_upload.md:11]:
```suggestion
Infinite Scale supports the tus resumable-upload protocol.
```

File: docs/apis/http/tus_upload.md [hunk 32]
```diff
@@ -0,0 +1,189 @@
...
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+The server will return a temporary upload URL in the location header of the response:
+```
+< Location: https://ocis.test/data/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJyZXZhIiwiZXhwIjoxNjk3NTMyNTc5LCJpYXQiOjE2OTc0NDYxNzksInRhcmdldCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTE1OC9kYXRhL3R1cy8zYTU3ZWZlMS04MzE0LTQ4MGEtOWY5Ny04N2Q1YzBjYTJhMTgifQ.FbrlY7mdOfsbFgMrP8OtcHlCEq72a2ZVnPD2iBo9MfM
+```
+
+### Upload the First Chunk
```
Comment (mmattel) [inline docs/apis/http/tus_upload.md:70]:
```suggestion
### Upload the First Chunk

```

File: docs/apis/http/tus_upload.md [hunk 33]
```diff
@@ -0,0 +1,189 @@
...
+< Vary: Origin
+< X-Content-Type-Options: nosniff
+< X-Request-Id: xxxxxxxxxxxxxxxxxxxxxx
+<
+* Connection #0 to host localhost left intact
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+### Upload Further Chunks
```
Comment (mmattel) [inline docs/apis/http/tus_upload.md:100]:
```suggestion
### Upload Further Chunks

```

File: docs/apis/http/tus_upload.md [hunk 34]
```diff
@@ -0,0 +1,189 @@
...
+* Connection #0 to host localhost left intact
+```
+{{< /tab >}}
+{{< /tabs >}}
+{{< hint type=warning title="Important Warning" >}}
+`Upload-Offset` header indicates the byte position in the target file where the server should start writing the upload content.
+It ensures data integrity and order during the upload process.
+{{< /hint >}}
+
+## Creation with Upload
```
Comment (mmattel) [inline docs/apis/http/tus_upload.md:136]:
```suggestion
## Creation with Upload

```

**Comment by mmattel** [comment]
@amrita-shrestha only some small improvements. You can collect them to a block and commit them at once.

**Review by individual-it — APPROVED** [review]

**Review by mmattel — APPROVED** [review]
